### PR TITLE
fix(describeTable): MAX string length for mssql

### PR DESCRIPTION
**Pull request by UziTech**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

MSSQL got support for varchar lengths in #9896 but if the length is `MAX` in the database the length === `-1`

This PR will check for that case and change the length to `MAX`

### CODE REVIEW

**Comment by UziTech** [comment]
I'm not exactly sure how to write a test for this.

**Comment by mirkojotic** [comment]
Hi @UziTech 
I would make a new test case in this [file](https://github.com/sequelize/sequelize/blob/4cc7dc8a4d9b85dc7dfef5064175a3cdc09c0fa8/test/integration/query-interface/describeTable.test.js#L59) and make a condition to check if the dialect is `mssql` [like so](https://github.com/sequelize/sequelize/blob/4cc7dc8a4d9b85dc7dfef5064175a3cdc09c0fa8/test/integration/dialects/mssql/regressions.test.js#L10). Just define one field with `MAX` length, perform a `describeTable` and check it is correct. Let me know if you need anything :wink:

File: test/integration/dialects/mssql/regressions.test.js [hunk 1]
```diff
@@ -82,4 +82,17 @@ if (dialect.match(/^mssql/)) {
         });
     });
   });
+
+  it('sets the varchar(max) length correctly on describeTable', function() {
```
Comment (sushantdhiman) [inline test/integration/dialects/mssql/regressions.test.js]: Perhaps `dialect === 'mssql'` check will be required, I don't think `VARCHAR(MAX)` is supported by other dialects.

Comment (UziTech) [reply test/integration/dialects/mssql/regressions.test.js]: these tests are wrapped in

```js
if (dialect.match(/^mssql/)) {
```

**Comment by UziTech** [comment]
Finally got it working 😃 🎉

**Pull request merged**